### PR TITLE
Fix to support filenames with periods in hiera 3

### PR DESCRIPTION
**Pull request by syntacticvexation**

### CODE REVIEW

File: lib/hiera/backend/file_backend.rb [hunk 1]
```diff
@@ -13,8 +13,9 @@ def initialize
         end
       end
 
-      def lookup(key, scope, order_override, resolution_type)
+      def lookup(key, scope, order_override, resolution_type, context)
```
Comment (igalic) [inline lib/hiera/backend/file_backend.rb]: this is a contract change.
can we provide a default to make this function backwards compatible?

Comment (jyaworski) [inline lib/hiera/backend/file_backend.rb]: I agree here.

**Comment by igalic** [comment]
@syntacticvexation ping

**Comment by syntacticvexation** [comment]
I've upgraded my environment to Hiera 3.  Can you advise whether it works as is under Hiera 1?

**Comment by buzzdeee** [comment]
I ran into the same issue, evaluating puppet 4 with hiera3.

I tested the patch with hiera 1.3.4, hiera-file 1.1.1, and puppet 3.8.7, and got this error:

Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Evaluation Error: Error while evaluating a Function Call, wrong number of arguments (4 for 5) at /etc/puppet/environments/production/manifests/site.pp:21:19 on node galen.l00-bugdead-prods.de
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run

**Comment by syntacticvexation** [comment]
Yep, sounds like a function arity problem.  I'll take a look late next week when I have some time to configure up hiera 1 again.

**Comment by syntacticvexation** [comment]
I've updated the backend file to hopefully address both Hiera 1 and Hiera 3.  Note that Hiera 2 to Hiera 3.0.5 does not support periods at all due to the change in how periods were used but without the extra lookup segment function to emulate the previous behaviour.  In this instance, a warning message is issued.

**Comment by bastelfreak** [comment]
Hi @syntacticvexation, are you still interested in getting this merged? If so, could you please rebase and take a look at the travis failures?

**Pull request closed without merging**